hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/upgrade/EightToSevenDowngradeHandler.java

```diff
@@ -92,4 +116,57 @@ private static void downgradePartitionFields(HoodieWriteConfig config, HoodieEng
       tablePropsToAdd.put(HoodieTableConfig.PARTITION_FIELDS, tableConfig.getPartitionFieldProp());
     }
   }
+
+  static void downgradeMetadataPartitions(HoodieEngineContext context,
+                                          HoodieStorage hoodieStorage,
+                                          HoodieTableMetaClient mdtMetaClient,
+                                          Map<ConfigProperty, String> tablePropsToAdd) {
+    // Fetch metadata partition paths.
+    List<String> metadataPartitions = FSUtils.getAllPartitionPaths(
+        context,
+        hoodieStorage,
+        mdtMetaClient.getBasePath(),
+        false);
+
+    // Delete partitions.
+    List<String> validPartitionPaths =
+        deleteMetadataPartition(context, mdtMetaClient, metadataPartitions);
+
+    // Clean the configuration.
+    tablePropsToAdd.put(
+        TABLE_METADATA_PARTITIONS, String.join(",", validPartitionPaths));
+  }
+
+  static List<String> deleteMetadataPartition(HoodieEngineContext context,
+                                              HoodieTableMetaClient mdtMetaClient,
+                                              List<String> metadataPartitions) {
+    metadataPartitions.stream()
+        .filter(metadataPath -> !SUPPORTED_METADATA_PARTITION_PATHS.contains(metadataPath))
+        .forEach(metadataPath ->
+            HoodieTableMetadataUtil.deleteMetadataTablePartition(
+                mdtMetaClient, context, metadataPath, true)
```

Comment:
I think this method expects data table meta client - https://github.com/apache/hudi/blob/cecfbbddffd2d7bdf105523412aa7c84127f7cfd/hudi-common/src/main/java/org/apache/hudi/metadata/HoodieTableMetadataUtil.java#L1534

Reply:
Updated.